connectors/sources/sharepoint_online.py

```diff
@@ -879,258 +995,283 @@ async def validate_config(self):
             )
 
-    @cached_property
-    def _default_groups(self):
-        return [_prefix_group(default_group) for default_group in DEFAULT_GROUPS]
-
     def _decorate_with_access_control(self, document, access_control):
         if self._dls_enabled():
             document[ACCESS_CONTROL] = list(
-                set(
-                    document.get(ACCESS_CONTROL, [])
-                    + access_control
-                    + self._default_groups
-                )
+                set(document.get(ACCESS_CONTROL, []) + access_control)
             )
 
         return document
 
-    async def _with_site_access_control(self, site):
+    async def _site_access_control(self, site):
+        """Fetches all permissions for all owners, members and visitors of a given site.
+        All groups and/or persons, which have permissions for a given site are returned with their given identity prefix ("user", "group" or "email").
+        For the given site all groups and its corresponding members and owners (username and/or email) are fetched.
+
+        Returns:
+            list: access control list for a given site
+            [
+                "user:spo-user",
+                "email:some.user@spo.com",
+                "group:1234-abcd-id"
+            ]
+        """
+
         if not self._dls_enabled():
-            return site
+            return []
 
-        site_web_url = site["webUrl"]
+        access_control = set()
 
-        sharepoint_groups = []
+        async for user_information in self.client.user_information_list(site["id"]):
+            user = user_information["fields"]
 
-        async for sharepoint_group in self.client.site_groups(site_web_url):
-            sharepoint_groups.append(sharepoint_group)
+            if is_domain_group(user):
+                domain_group_id = _domain_group_id(user["Name"])
 
-        sharepoint_groups = list(
-            map(
-                lambda group: _prefix_group(group),
-                filter(
-                    lambda group_title: group_title is not None,
-                    map(
-                        lambda group: group.get("Title"),
-                        sharepoint_groups,
-                    ),
-                ),
-            )
-        )
+                if domain_group_id:
+                    access_control.add(_prefix_group(domain_group_id))
+
+            if is_person(user):
+                name = user.get("Name")
+
+                if name and name.startswith("i:0#.f|membership|"):
+                    parts = name.split("|")
+
+                    if len(parts) > 2:
+                        email = parts[2]
+                        access_control.add(_prefix_email(email))
+                        continue
+
+                if "EMail" not in user:
+                    continue
+
+                email = user["EMail"]
```

Comment:
It should be consistent, I see you use `user["Name"]` and `user.get("Name")` above without checking presence.